tests/components/tabs/Tab.test.js

```diff
@@ -8,13 +8,27 @@ describe('`Tab` component', () => {
     });
 
     it('should trigger event when clicked', () => {
+        const event = { preventDefault: () => {} };
+        jest.spyOn(event, 'preventDefault');
         const handleOnClick = jest.fn();
         const wrapper = shallow(<Tab onClick={handleOnClick}/>);
         const tab = wrapper.find('a');
         expect(handleOnClick.mock.calls.length).toBe(0);
-        tab.simulate('click', {
-            preventDefault: () => {},
-          });
+        tab.simulate('click', event);
         expect(handleOnClick.mock.calls.length).toBe(1);
     });
+    it('should trigger event when clicked and isActive prop set to true', () => {
+        const event = { preventDefault: () => {} };
+        jest.spyOn(event, 'preventDefault');
+        const props = {
+            isActive: true,
+            handleOnClick: jest.fn(),
+            tabName: 'random tab name',
+            tabIndex: 2,
+        }
+        const wrapper = shallow(<Tab {...props} />);
+        const tab = wrapper.find('a');
+        tab.simulate('click', event);
+        expect(props.handleOnClick.mock.calls.length).toBe(0);
```

Comment:
is this assertion not supposed to be

`expect(props.handleOnClick.mock.calls.length).toBe(1);` ??

the test-suite's message says that the `isActive` prop is set to true when the click event is triggered, but here you are setting the isActive prop key to true when the button hasn't been clicked. there is no assertion for the state change as well.

Reply:
resolved!